src/photon/photon_scheduler.c

```diff
@@ -114,7 +114,12 @@ void assign_task_to_worker(local_scheduler_state *state,
                            bool from_global_scheduler) {
   CHECK(worker_index < utarray_len(state->workers));
   worker *w = (worker *) utarray_eltptr(state->workers, worker_index);
-  write_message(w->sock, EXECUTE_TASK, task_spec_size(spec), (uint8_t *) spec);
+  if (write_message(w->sock, EXECUTE_TASK, task_spec_size(spec),
+                    (uint8_t *) spec) < 0) {
+    /* TODO(rkn): If this happens, the task should be added back to the task
+     * queue. */
```

Comment:
might make sense to just do it (add back to the queue) as part of this PR? You have the ``state``.